samples/nrf9160/modem_shell/src/link/link.c

```diff
@@ -327,15 +331,39 @@ void link_ind_handler(const struct lte_lc_evt *const evt)
 				cells.neighbor_cells[i].time_diff);
 		}
 
-		ncellmeas_work_data.mode = ncellmeas_mode;
-		ncellmeas_work_data.search_type = ncellmeas_search_type;
-		ncellmeas_work_data.periodic_interval = ncellmeas_periodic_interval;
-		if (ncellmeas_mode == LINK_NCELLMEAS_MODE_CONTINUOUS &&
-		    ncellmeas_periodic_interval) {
+		/* Surrounding cells from GCI search types */
+		for (i = 0; i < cells.gci_cells_count; i++) {
+			/* GCI neighbor cells: */
+			if (cells.gci_cells[i].timing_advance ==
+				LTE_LC_CELL_TIMING_ADVANCE_INVALID) {
+				sprintf(tmp_ta_str, "\"not valid\"");
+			} else {
+				sprintf(tmp_ta_str, "%d", cells.gci_cells[i].timing_advance);
+			}
+			mosh_print("  Surrounding cell %d from GCI search", i + 1);
+			mosh_print(
+				"    ID %d, phy ID %d, MCC %d MNC %d, RSRP %d : %ddBm, RSRQ %d, TAC %d, earfcn %d, meas time %lld\n"
+				"    TA %s, TA meas time %lld",
+				cells.gci_cells[i].id, cells.gci_cells[i].phys_cell_id,
+				cells.gci_cells[i].mcc, cells.gci_cells[i].mnc,
+				cells.gci_cells[i].rsrp,
+				RSRP_IDX_TO_DBM(cells.gci_cells[i].rsrp),
+				cells.gci_cells[i].rsrq, cells.gci_cells[i].tac,
+				cells.gci_cells[i].earfcn, cells.gci_cells[i].measurement_time,
+				tmp_ta_str,
+				cells.gci_cells[i].timing_advance_meas_time);
+		}
+
+		ncellmeas_work_data.mode = ncellmeas_param_data.mode;
+		ncellmeas_work_data.params.search_type = ncellmeas_param_data.params.search_type;
+		ncellmeas_work_data.params.gci_count = ncellmeas_param_data.params.gci_count;
+		ncellmeas_work_data.periodic_interval = ncellmeas_param_data.periodic_interval;
```

Comment:
If these are only used for the periodic mode, these could probably be moved inside the `if` below.

Reply:
true, moved.